pybamm/expression_tree/operations/evaluate_julia.py

```diff
@@ -0,0 +1,1101 @@
+                    data = np.round(data, decimals=11)
+                m, n = value.shape
+                # Set print options large enough to avoid ellipsis
+                # at least as big is len(row) = len(col) = len(data)
+                np.set_printoptions(
+                    threshold=max(np.get_printoptions()["threshold"], len(row) + 10)
+                )
+                # increase precision for printing
+                np.set_printoptions(precision=20)
+                # add 1 to correct for 1-indexing in Julia
+                # use array2string so that commas are included
+                constant_symbols[symbol.id] = "sparse({}, {}, {}, {}, {})".format(
+                    np.array2string(row + 1, separator=","),
+                    np.array2string(col + 1, separator=","),
+                    np.array2string(data, separator=","),
+                    m,
+                    n,
+                )
+            elif value.shape == (1, 1):
+                # Extract value if array has only one entry
+                constant_symbols[symbol.id] = value[0, 0]
+                variable_symbol_sizes[symbol.id] = 1
+            elif value.shape[1] == 1:
+                # Set print options large enough to avoid ellipsis
+                # at least as big as len(row) = len(col) = len(data)
+                np.set_printoptions(
+                    threshold=max(
+                        np.get_printoptions()["threshold"], value.shape[0] + 10
+                    )
+                )
+                # Flatten a 1D array
+                constant_symbols[symbol.id] = np.array2string(
+                    value.flatten(), separator=","
+                )
+                variable_symbol_sizes[symbol.id] = symbol.shape[0]
+            else:
+                constant_symbols[symbol.id] = value
+                # No need to save the size as it will not need to be used
+        return
+
+    # process children recursively
+    for child in symbol.children:
+        find_symbols(
+            child,
+            constant_symbols,
+            variable_symbols,
+            variable_symbol_sizes,
+            round_constants=round_constants,
+        )
+
+    # calculate the variable names that will hold the result of calculating the
+    # children variables
+    children_vars = []
+    for child in symbol.children:
+        if isinstance(child, pybamm.Broadcast):
+            child = child.child
+        if is_constant_and_can_evaluate(child):
+            child_eval = child.evaluate()
+            if isinstance(child_eval, numbers.Number):
+                children_vars.append(str(child_eval))
+            else:
+                children_vars.append(id_to_julia_variable(child.id, "const"))
+        else:
+            children_vars.append(id_to_julia_variable(child.id, "cache"))
+
+    if isinstance(symbol, pybamm.BinaryOperator):
+        # Multiplication and Division need special handling for scipy sparse matrices
```

Comment:
is this comment current? I can't see any special handling for scipy sparse matrices